lib/fluent/plugin/filter_grep.rb

```diff
@@ -54,65 +59,157 @@ def initialize
       config_param :key, :string
       desc "The regular expression."
       config_param :pattern do |value|
-        if value.start_with?("/") and value.end_with("/")
+        if value.start_with?("/") and value.end_with?("/")
           Regexp.compile(value[1..-2])
         else
           Regexp.compile(value)
         end
       end
     end
 
-    # for test
-    attr_reader :_regexps
-    attr_reader :_excludes
+    config_section :and, param_name: :and_conditions, multi: true do
+      config_section :regexp, param_name: :regexps, multi: true do
+        desc "The field name to which the regular expression is applied."
+        config_param :key, :string
+        desc "The regular expression."
+        config_param :pattern do |value|
+          if value.start_with?("/") and value.end_with?("/")
```

Comment:
Hmm... I see these patterns, we need to `:regexp` type for config_param.

@okkez Could you write a patch for adding `regexp` type in other PR?

Reply:
Sure, I will write a patch for this.

Reply:
I've created #1927